proposals/NNNN-modify-and-read-accessors.md

```diff
@@ -0,0 +1,621 @@
+# Modify and read accessors
+
+* Proposal: [SE-NNNN](NNNN-modify-and-read-accessors.md)
+* Authors: [Ben Cohen](https://github.com/airspeedswift), [Nate Chandler](https://github.com/nate-chandler)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation: Partially available on main behind the frontend flag `-enable-experimental-feature CoroutineAccessors`
+* Previous Pitch: https://forums.swift.org/t/modify-accessors/31872
+
+## Introduction
+
+We propose the introduction of two new keywords--`modify` and `read`--for implementing computed properties and subscripts, alongside the current `get` and `set`.
+
+The body of a `modify` or `read` implementation will be a coroutine, and it will introduce a new contextual keyword, `yield`, that will be used to lend a potentially mutable value back to the caller as the coroutine runs.
+When the caller resumes the coroutine, its execution will continue from after that `yield`.
+
+These coroutine accessors enable values to be accessed and changed without requiring a copy.
+This is essential for noncopyable types and generally desirable elsewhere for performance.
+
+This feature has been available (but not supported) since Swift 5.0 via the `_modify` and `_read` keywords.
+Additionally, the feature is available via `read` and `modify` on recent main with the flag `-enable-experimental-feature CoroutineAccessors`.
+
+## Motivation
+
+### Modify<a name="modify-motivation"/>
+
+Swift's `get`/`set` syntax allows users to expose computed properties and subscripts that behave as l-values.
+This powerful feature allows for the creation of succinct idiomatic APIs, such as this use of `Dictionary`'s defaulting subscript:
+
+```swift
+var wordFrequencies: [String:Int] = [:]
+wordFrequency["swift", default: 0] += 1
```

Comment:
I presume `wordFrequency` here is a typo and should be `wordFrequencies`?

Reply:
Thanks, fixed.